### ISSUE TITLE
custom content on mob whith mythicmobs

### ISSUE THREAD

**Issue by WhiteC4D** [PluginBugs/Issues-ItemsAdder#4002]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I've looked for already existing issues on the [Issue Tracker](https://github.com/PluginBugs/Issues-ItemsAdder/issues) and haven't found any.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.
- [X] I tested this with just ItemsAdder and its dependencies and with a vanilla client of the same version as the Server.

### Discord Username (optional)

.raven_white

### What happened?

in my mob configs on mythicmobs, I therefore equipped my obs with armor and itemsadder sword, but when it spawns, no armor or sword, just like any item. This works before a version change of itemsadder and all our plugins

the wiki
![image](https://github.com/user-attachments/assets/71fe60fe-3a59-4557-8f64-d20fb3cf47d3)

my config
![image](https://github.com/user-attachments/assets/cc8dbf28-bba4-4974-bcfc-7d2d8fe42a68)

### Steps to reproduce the issue

equip mobs with armor on a mob's config in Equipment

### Server version

This server is running UniverseSpigot version 1.21.1-DEV-ver/1.21.1@014ae00 (2024-09-08T01:48:53Z) (Implementing API version 1.21.1-R0.1-SNAPSHOT)
Universe-1.21.1-DEV-014ae00 (MC: 1.21.1)

### ItemsAdder Version

4.0.2-beta-release-10

### ProtocolLib Version

5.3.0-SNAPSHOT-726

### LoneLibs Version

1.0.58

### Full Server Log

https://mclo.gs/uj86caf

### Error (optional)

```shell
[17:56:21 INFO]: whitec4d issued server command: /mm mobs spawn D2_déchu5
[17:56:21 ERROR]: [MythicMobs] [SCHEDULER] Exception thrown whilst executing task
[17:56:21 WARN]: java.lang.InstantiationError: io.lumine.mythic.bukkit.adapters.BukkitItemStack
[17:56:21 WARN]:        at ItemsAdder-4.0.2_beta_release_10.jar//dev.lone.itemsadder.Core.OtherPlugins.MythicMobs.drops.CustomItemDrop.getDrop(SourceFile:28)
[17:56:21 WARN]:        at MythicMobs-5.7.1.jar//io.lumine.mythic.core.drops.LootBag.equip(LootBag.java:239)
[17:56:21 WARN]:        at MythicMobs-5.7.1.jar//io.lumine.mythic.core.mobs.MobType.lambda$applyMobOptions$5(MobType.java:1150)
[17:56:21 WARN]:        at MythicMobs-5.7.1.jar//io.lumine.mythic.bukkit.utils.Delegates$RunnableToSupplier.get(Delegates.java:121)
[17:56:21 WARN]:        at MythicMobs-5.7.1.jar//io.lumine.mythic.bukkit.utils.promise.LuminePromise$SupplyRunnable.run(LuminePromise.java:654)
[17:56:21 WARN]:        at MythicMobs-5.7.1.jar//io.lumine.mythic.bukkit.utils.tasks.LumineExecutors$SchedulerWrappedRunnable.run(LumineExecutors.java:70)
[17:56:21 WARN]:        at org.bukkit.craftbukkit.scheduler.CraftTask.run(CraftTask.java:86)
[17:56:21 WARN]:        at org.bukkit.craftbukkit.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:475)
[17:56:21 WARN]:        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1740)
[17:56:21 WARN]:        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:489)
[17:56:21 WARN]:        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1612)
[17:56:21 WARN]:        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1312)
[17:56:21 WARN]:        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:335)
[17:56:21 WARN]:        at java.base/java.lang.Thread.run(Thread.java:1583)
```

**Comment by WhiteC4D** [comment]
> @LoneDev6 If you use the method called "io.lumine.mythic.bukkit.BukkitAdapter.adapt(Itemstack Item)", it returns BukkitItemStack. Instead of this, "io.lumine.mythic.bukkit.adapters.BukkitItemStack".
> 
> ![image](https://private-user-images.githubusercontent.com/55119171/366975459-d7042742-7911-4c80-8643-fa645cc2f91f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzMzg1NTksIm5iZiI6MTcyNjMzODI1OSwicGF0aCI6Ii81NTExOTE3MS8zNjY5NzU0NTktZDcwNDI3NDItNzkxMS00YzgwLTg2NDMtZmE2NDVjYzJmOTFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDE4MjQxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2Y2FlNjYyY2ZiZWQ4Njc4OWQ1NGVmMTc4ZTdjZjdjNzdlMGY2NzlhYTE5MmE1NDM3MTgxMmM3ZTIwZTMwYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KRBrhEm6Bt1ACbmbmBhgrN5Ds4lG4kzXEu_rQuxzUO4)
> 
> I'm struggling with incompatibility... actually, I'll appreciate it if you deploy the fixed version, please....
> 
> This will fix #3915, #3879 together

how you find this ?

**Comment by WhiteC4D** [comment]
@yhdev7935

**Comment by yhdev7935** [comment]
@WhiteC4D MythicMobs changes their identifier of the method into 'protected'. It means it's not able to approach the method by instantiation like that.
For that reason, you have to open the MythicMob plugin using java decompiler which you can easily find on any websites.
That's it. and I found the alternative method. :3

**Comment by KeithR77** [comment]
@yhdev7935 Were you able to find some sort of workaround to get MythicMob/IA integration to work by decompiling, modifying the code, and recompiling? Or is this something we'd still have to wait on LoneDev to update and fix..

**Comment by Florenthz** [comment]
Hi, I have the same problem, has LoneDev said something?

**Comment by KeithR77** [comment]
> Hi, I have the same problem, has LoneDev said something?

He responded to the #3915 incident, but that's about it. There's been 4 new releases/patches for ItemsAdder since he responded to that incident, but none of them have included a fix for MythicMobs compatibility. All he did was move the #3915 ticket to the 'future-updates' milestone, and it's been sitting there ever since unresolved. My original issue/ticket #3879 about MythicMobs compatibility was opened even farther back (6 weeks ago...).

**Comment by Florenthz** [comment]
I see, I will post the problem too, maybe he give more attention to the issue, greetings friend :D

**Comment by ImReverseV** [comment]
@LoneDev6 @Andre601 got any news about this? I've experienced this issue for a month and a half now, but didn't want to open an issue since many people already did that.

**Comment by LoneDev6** [comment]
Duplicate of https://github.com/PluginBugs/Issues-ItemsAdder/issues/3879
